### ISSUE TITLE
Using PEP 604 to allow unions written as x | y in Python 3.9

### ISSUE THREAD

**Issue by HansAarneLiblik** [marshmallow-code/marshmallow#2066]
Currently due to Python 3.10 not being available for us, we're stuck with Python 3.9

Your source code currently uses [PEP 604](https://peps.python.org/pep-0604/) which makes errors for IDE when running with Python 3.9

https://github.com/marshmallow-code/marshmallow/blob/dev/src/marshmallow/fields.py#L160-L164

![image](https://user-images.githubusercontent.com/5418540/199802374-9e5bfe14-ce93-4823-9cdb-694a0e5c3f93.png)

We're using `pipenv` as our environment manager and have installed `marshmallow = "==3.18.*"`

**Comment by HansAarneLiblik** [comment]
@lafrech Any ideas or progress on this ?

@Isira-Seneviratne You've made these changes to the code, we're you aware of these side-effects ?

**Comment by lafrech** [comment]
This line at the top of the file backports PEP 604 to Python 3.9 so it should work:

```py
from __future__ import annotations
```

BTW, we still use `Union` in a few places. We might want to use a pipe instead. (See #2071)

**Comment by HansAarneLiblik** [comment]
Aha, it seems its due to line-breaks on union

```py
def __init__(
  self,
  ...
  validate: None | (typing.Callable[[typing.Any], typing.Any] | typing.Iterable[typing.Callable[[typing.Any], typing.Any]]) = None
  ...
) -> None:
```

works perfectly (Did not linebreak on `|`)

**Comment by lafrech** [comment]
OK, so it's an issue with the linter or the editor, I guess.

Closing this as not a marshmallow issue. Feel free to comment further.

**Comment by dpmccabe** [comment]
Looks like this got fixed in 8168ef2a6d18b8bd33b1ad9eaa17fddb0f2f9547

**Comment by HansAarneLiblik** [comment]
This change seems to work, when I made it locally. 

Any idea when it will be released? @lafrech
